clients/mtaclient/mta_rest_client.go

```diff
@@ -255,7 +258,11 @@ func (c MtaRestClient) UploadMtaFile(file util.NamedReadSeeker, fileSize int64,
 func (c MtaRestClient) calculateRequestSize(fileName string, fileSize int64) (int64, error) {
 	var body bytes.Buffer
 	form := multipart.NewWriter(&body)
-	_, err := form.CreateFormFile("file", fileName)
+	h := make(textproto.MIMEHeader)
+	h.Set("Content-Disposition", fmt.Sprintf(`form-data; name="file"; filename="%s"`, escapeQuotes(fileName)))
+	h.Set("Content-Type", "application/octet-stream")
+	h.Set("Content-Length", strconv.FormatInt(fileSize, 10))
```

Comment:
What will be difference when `Content-Legth` is set?

Reply:
If Tomcat was smart, it would read this length and set it in the `Part` object as its size, instead of persisting the entire file to disk.